### PR TITLE
qemu: Only enable uefi related options when we want to boot using uefi

### PR DESCRIPTION
**Pull request by DaanDeMeyer**

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -7715,12 +7715,20 @@ def qemu_check_kvm_support() -> bool:
 def run_qemu_cmdline(args: MkosiArgs) -> Iterator[List[str]]:
     accel = "kvm" if args.qemu_kvm else "tcg"
 
+    if "uefi" in args.boot_protocols:
+        mode = "uefi"
+    elif "bios" in args.boot_protocols:
+        mode = "bios"
+    else:
+        mode = "uefi"
```
Comment (behrmann) [inline mkosi/__init__.py:7723]:
```suggestion
    mode = "bios" if args.boot_protocols == ["bios"] else "uefi"
```
We ensure early on that `args.boot_protocols` can at most be `["bios", "uefi"]` (in some order).

**Review by behrmann — COMMENTED** [review]
LGTM, if this fixes CI.

File: mkosi/__init__.py (hunk 1 above)
Comment (DaanDeMeyer) [reply mkosi/__init__.py:7723]: Yeah but when we add the linux boot we'll need to split it up again so I would keep it as is

**Comment by behrmann** [comment]
Still has SSH connection errors on Fedora and Gentoo `gpt_ext4`, but at least it boots to prompt. Gonna merge this so that the other `gpt_ext4` CI failures are fixed.

**Pull request merged**